server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/dto/DLPConfigurationDTO.java

```diff
@@ -0,0 +1,140 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.dto;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.apache.james.dlp.api.DLPConfigurationItem;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.github.steveash.guavate.Guavate;
+import com.google.common.base.Preconditions;
+
+public class DLPConfigurationDTO {
+
+    public static List<DLPConfigurationItem> toDLPConfigurations(List<DLPConfigurationDTO> dtos) {
+        Preconditions.checkNotNull(dtos);
+
+        return dtos.stream()
+            .map(DLPConfigurationDTO::toDLPConfiguration)
+            .collect(Guavate.toImmutableList());
+    }
+
+    public static List<DLPConfigurationDTO> toDTOs(List<DLPConfigurationItem> dlpConfigurations) {
+        Preconditions.checkNotNull(dlpConfigurations);
+
+        return dlpConfigurations.stream()
+            .map(DLPConfigurationDTO::toDTO)
+            .collect(Guavate.toImmutableList());
+    }
+
+    private static DLPConfigurationItem toDLPConfiguration(DLPConfigurationDTO dto) {
+        return DLPConfigurationItem.builder()
+            .id(DLPConfigurationItem.Id.of(dto.id))
```

Comment:
I don't like this because the meaning is that as a user I'm forced to supply an Id.

Maybe this Id should be generated by James?

Reply:
If ID should be generated in James, then it's better to generate ID inside `DLPConfigurationStore`, not in webadmin, WDYT? And what kind of ID you might want? UUID?